src/axom/core/ArrayView.hpp

```diff
@@ -371,7 +371,7 @@ ArrayView<T, DIM, SPACE>::ArrayView(
   , m_allocator_id(static_cast<const OtherArrayType&>(other).getAllocatorID())
 {
   static_assert(
-    std::is_const<T>::value,
+    std::is_const<T>::value || detail::ArrayTraits<OtherArrayType>::is_view,
```

Comment:
Why do we not allow `other` to be a view?  Do we want to exclude views of views?  Or does it have something to do with the const-ness of the data?

Reply:
The latter -- this constructor matches with `const` instances of `Array`, so we want to prevent it from being called to construct a mutable `ArrayView<T>`. But we can relax this requirement for view-like types, such as `ArraySubslice`. This is used in the Slam multi-dimensional indexers to return sub-ranges.